### DISCUSSION TITLE
v1.0.7

### DISCUSSION THREAD

**Discussion by giuseppe99barchetta** [giuseppe99barchetta/SuggestArr#29] [Changelog]
### 🚨 Important
- **Plex Support Integration**: Added support for Plex! Allowing users to login directly with their Plex account, retrieve server and libraries, and integrate content requests.

### ⚠️ Breaking Changes
- **Edited environment variables name**:  Some environment variables have been renamed. Please review your configuration to ensure it remains valid. If any issues arise, it's recommended to reset your configuration and start fresh with the new variable names.

### ✨ Added
- **New step for select the media Service**: The first step now allows user to select which media server want to use (Plex or Jellyfin).
- **Real-Time Log Panel with Filters**: Added a real-time log viewing panel with multiple filters (e.g., `INFO`, `ERROR`, `DEBUG`), allowing users to filter and view application logs directly from the UI for better diagnostics.
- **Implemented **: Implemented UI for testing Plex API connections directly from the configuration page.
- **Trailing Slash Handling**: Automatically removes trailing slashes from URLs in Jellyfin, Jellyseer and Plex configurations to ensure consistency and avoid API errors.
- **Unified Seer Management**: Implemented support for both Jellyseer and Overseer under unified variables for easier configuration (`SEER_API_URL`, `SEER_TOKEN`, etc.).
- **Background Image Rotation**: Added dynamic background image rotation in the configuration wizard to improve visual experience.
- **Default Background Images**: Added fallback background images in the wizard until a valid TMDB API Key is provided.
- **Configuration Pre-testing and Pre-authentication**: Automatically checks if configuration (e.g., API keys, URLs) already exists and marks them as validated and authenticated, skipping manual re-testing steps if previously successful.
  
### 🚀 Improved
- **UI Enhancements**: Improved visual feedback in the form of spinners, validation icons, and error handling during API testing for both Jellyfin and Plex.
- **Code Modularization**: Refactored content automation logic into separate classes for Jellyfin and Plex, improving maintainability and readability.

<hr /><em>This discussion was created from the release <a href='https://github.com/giuseppe99barchetta/SuggestArr/releases/tag/v1.0.7'>v1.0.7</a>.</em>

**Comment by aizazmueez** [comment]
hey i having trouble getting my overseer API key recognized. I get the following error in the log

2024-10-24 20:19:48,427 - SeerClient - ERROR - Request to http://192.168.1.200:5055/api/v1/user timed out.

**Comment by giuseppe99barchetta** [comment]
> hey i having trouble getting my overseer API key recognized. I get the following error in the log
> 
> 
> 
> 2024-10-24 20:19:48,427 - SeerClient - ERROR - Request to http://192.168.1.200:5055/api/v1/user timed out.
> 
> 

Hi! Thank you for contacting me!

That’s really strange, I’ve never seen an error like this before. Have you double-checked the Overseer URL and API key?

At which step are you encountering this error?

**Reply by aizazmueez** [reply]
Step 4/5 Overseer API details.

My overseer is hosted at 192.168.1.200:5055 so thats the overseer url that i use. And the api key i have double checked.

**Reply by giuseppe99barchetta** [reply]
> Step 4/5 Overseer API details.
> 
> 
> 
> My overseer is hosted at 192.168.1.200:5055 so thats the overseer url that i use. And the api key i have double checked.

Mhh really strange, I can't reproduce it. Overseer is in a docker container?

Could you please try to recreate the container of SuggestArr?

**Reply by aizazmueez** [reply]
yes overseer is in a docker container.

**Reply by giuseppe99barchetta** [reply]
> yes overseer is in a docker container.
> 
> 
> 
> 

Let me know if after recreating the container the error persists.

If the error persists, could you please open an issue? That way, I can notify you once the problem is resolved.
